src/main/java/com/google/cloud/tools/gradle/endpoints/framework/server/task/EndpointsArtifactTask.java

```diff
@@ -110,10 +116,62 @@ public void setBasePath(String basePath) {
     this.basePath = basePath;
   }
 
+  @Internal
+  public boolean isCleanBeforeRun() {
+    return cleanBeforeRun;
+  }
+
+  public void setCleanBeforeRun(boolean cleanBeforeRun) {
+    this.cleanBeforeRun = cleanBeforeRun;
+  }
+
+  @Internal
+  public String getOutputLanguage() {
+    return outputLanguage;
+  }
+
+  public void setOutputLanguage(String outputLanguage) {
+    this.outputLanguage = outputLanguage;
+  }
+
+  @Internal
+  public String getOutputBuildSystem() {
+    return outputBuildSystem;
+  }
+
+  public void setOutputBuildSystem(String outputBuildSystem) {
+    this.outputBuildSystem = outputBuildSystem;
+  }
+
+  @Internal
+  public String getOutputFileName() {
+    return outputFileName;
+  }
+
+  public void setOutputFileName(String outputFileName) {
+    this.outputFileName = outputFileName;
+  }
+
   /** Task entry point. */
   @TaskAction
   void generateEndpointsArtifact() throws Exception {
 
+    if (cleanBeforeRun) {
+      getProject().delete(outputDirectory);
+      getProject().mkdir(outputDirectory);
+    }
+
+    List<String> params = new ArrayList<>();
+
+    params.add(command);
+
+    params.add("-o");
+    if (!Strings.isNullOrEmpty(outputFileName)) {
+      params.add(new File(outputDirectory, outputFileName).getAbsolutePath());
```

Comment:
This is using `getAbsolutePath` whereas the original was using `getPath`. I guess this is safer?

Reply:
Yeah I can't remember why I would have used getPath initially (I'm used to using getAbsolutePath), I'll change the -w to use getAbsolutePath too.